src/pymor/models/basic.py

```diff
@@ -87,6 +87,46 @@ def __str__(self):
     def _compute_solution(self, mu=None, **kwargs):
         return self.operator.apply_inverse(self.rhs.as_range_array(mu), mu=mu)
 
+    @property
+    def dual(self):
+        try:
+            return self._dual
+        except AttributeError:
+            assert self.output_functional is not None
+            assert self.output_functional.linear
+            # TODO: assert that the operator is symmetric
```

Comment:
Before merging this, we need to find a solution here. One simple option would be to let the user pass an optional `dual_operator`. However, I'm unsure if there are situations where also `output_functional` needs to be modified to be usable as a rhs.

Reply:
I mean, it is not even clear that a dual model has the `output_functional` as right hand side. This probably is something that is usual for RB or Optimization, but in general one could also go for the dual model with the actual rhs as its right hand side. 

We could thus also introduce an optional `dual_rhs` argument